target_chains/ethereum/contracts/contracts/pulse/IPulse.sol

```diff
@@ -30,7 +52,8 @@ interface IPulse is PulseEvents {
      *      the fee estimation unreliable.
      */
     function requestPriceUpdatesWithCallback(
-        uint256 publishTime,
+        address provider,
```

Comment:
I don't like people choosing their providers. It's an extra hop for integrators to think about (what are providers? how they are different? how should i choose? ...)

Reply:
Yeah agreed. I think the thing to do is to add a wrapper method that chooses for users. I'm not exactly sure what the right way to choose is yet though.